heron/healthm/src/java/com/twitter/heron/healthmgr/HealthManager.java

```diff
@@ -0,0 +1,386 @@
+    for (String policyId : policyIds) {
+      Map<String, String> policyConfigMap = policyConfigReader.getPolicyConfig(policyId);
+      HealthPolicyConfig policyConfig = new HealthPolicyConfig(policyConfigMap);
+
+      String policyClassName = policyConfig.getPolicyClass(policyId);
+      LOG.info(String.format("Initializing %s with class %s", policyId, policyClassName));
+      Class<IHealthPolicy> policyClass
+          = (Class<IHealthPolicy>) this.getClass().getClassLoader().loadClass(policyClassName);
+
+      AbstractModule module = constructPolicySpecificModule(policyConfig);
+      IHealthPolicy policy = injector.createChildInjector(module).getInstance(policyClass);
+
+      healthPolicies.add(policy);
+    }
+  }
+
+  @VisibleForTesting
+  HealthPolicyConfigReader createPolicyConfigReader() throws FileNotFoundException {
+    String policyConfigFile
+        = Paths.get(Context.heronConf(config), HealthMgrConstants.CONF_FILE_NAME).toString();
+    HealthPolicyConfigReader configReader = new HealthPolicyConfigReader(policyConfigFile);
+    configReader.loadConfig();
+    return configReader;
+  }
+
+  private AbstractModule constructConfigModule(final String trackerUrl) {
+    return new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(String.class)
+            .annotatedWith(Names.named(HealthMgrConstants.CONF_TRACKER_URL))
+            .toInstance(trackerUrl);
+        bind(String.class)
+            .annotatedWith(Names.named(HealthMgrConstants.CONF_TOPOLOGY_NAME))
+            .toInstance(Context.topologyName(config));
+        bind(String.class)
+            .annotatedWith(Names.named(HealthMgrConstants.CONF_CLUSTER))
+            .toInstance(Context.cluster(config));
+        bind(String.class)
+            .annotatedWith(Names.named(HealthMgrConstants.CONF_ENVIRON))
+            .toInstance(Context.environ(config));
+        bind(Config.class).toInstance(config);
+        bind(EventManager.class).in(Singleton.class);
+
+        bind(ISchedulerClient.class).toInstance(schedulerClient);
+        bind(SchedulerStateManagerAdaptor.class).toInstance(stateMgrAdaptor);
+        bind(MetricsProvider.class).to(TrackerMetricsProvider.class).in(Singleton.class);
+        bind(PackingPlanProvider.class).in(Singleton.class);
+      }
+    };
+  }
+
+  private AbstractModule constructPolicySpecificModule(final HealthPolicyConfig policyConfig) {
+    return new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(HealthPolicyConfig.class).toInstance(policyConfig);
+      }
+    };
+  }
+
+  @VisibleForTesting
+  SchedulerStateManagerAdaptor createStateMgrAdaptor() throws ReflectiveOperationException {
+    String stateMgrClass = Context.stateManagerClass(config);
+    IStateManager stateMgr = ReflectionUtils.newInstance(stateMgrClass);
+    stateMgr.initialize(config);
+    return new SchedulerStateManagerAdaptor(stateMgr, 5000);
```

Comment:
The state mgr needs to be initialized before it could be used in the adaptor. I can think of instantiating the stateMgr through guice, but not sure if adds any value. Using `ReflectionUtils` is consistent with the usage in other places.